modules/mod-deep-learning/DeepModelManager.h

```diff
@@ -0,0 +1,135 @@
+/**********************************************************************
+
+   Audacity: A Digital Audio Editor
+   Audacity(R) is copyright (c) 1999-2021 Audacity Team.
+
+   DeepModelManager.h
+   Hugo Flores Garcia
+
+******************************************************************/
+/**
+
+\class DeepModelManager
+\brief tools for downloading and managing Audacity models hosted in HuggingFace
+
+*/
+/*******************************************************************/
+
+#pragma once
+
+#include "DeepModel.h"
+#include "ModelCard.h"
+
+#include "Request.h"
+#include "NetworkManager.h"
+#include "widgets/ProgressDialog.h"
+#include "AudacityException.h"
+
+// this exception should be caught internally, but we 
+// derive from MessageBoxException just in case it needs to 
+// get handled by Audacity
+class ModelManagerException final : public MessageBoxException
```

Comment:
This is the third time I'm reading a similar new exception class, defining `what()`.  Perhaps the common pattern needs to be a new utility I should add to lib-exceptions.